scala-thoughts.md

```diff
@@ -0,0 +1,37 @@
+### What's easy to do in Scala? What's not?
+* I was impressed with the variety of methods on lists! I used map and filter, because those
+were methods that I learned in my CS52 class last semester. 
+One of Scala's greatest strengths is how concise it can be,
+after you learn how to properly chain methods. 
+
+### What is/are your favorite language design choice(s) that the designers of Scala made? Why?
+* To be honest, I found this assignment to be quite difficult... but here are some of the cool things that I observed.
+* I don't know whether I like this, but I think it's interesting that Scala is designed so that a "phrase of code" doesn't need
+to end in a semicolon. I think it's nice that having it doesn't hurt. Maybe this is because Scala was designed to cater to the 
+coding habits of a lot of different users who are used to different programming languages.
+* Even though I had to get used to Scala's way of constructing method headers, I think I like them! For example,
+def isPalindrome(str: String): Boolean = { ... } makes sense to me. I especially like "Boolean = { ... }" because it makes it 
+clear that whatever is returned by the code between the curly brackets is of type Boolean. 
+
+### What is/are your least favorite language design choice(s)? Why? And why do you think the designers made that / those choice(s)?
+* One of my least favorite design choices is how to print a string. Perhaps it's because I'm used to how it's done in Java, 
+but typing the following: s"A string with length ${str.length}" was so tedious! I much prefer writing "A string with length " 

```

Comment:
Actually, Scala allows both string interpolation and concatenation (yay options). Other languages like Ruby use string interpolation, too, so it's not unique to Scala. 

One reason why Scala might include interpolation is because in some languages (Python), you can only concatenate strings and not other types. Someone probably thought writing `str` every time is tedious, so string interpolation allows programmers to insert non-strings into strings.
```
s = "This integer is " + str(4)
```